File: examples/laravel/app/Http/Controllers/KomaController.php
<?php

namespace App\Http\Controllers;

use App\Services\KomaService;
use Illuminate\Http\Request;

class KomaController extends Controller
{
    public function __construct(private KomaService $koma) {}

    public function qr(Request $request)
    {
        $body = $request->validate([
            'amount' => ['required', 'string'],
            'currency' => ['required', 'string'],
            'productId' => ['required', 'string'],
        ]);

        return response()->json(
            $this->koma->createQrSession($body['amount'], $body['currency'], $body['productId'])
        );
    }

    public function status(Request $request)
    {
        return response()->json($request->validate([
            'md5' => ['required', 'string'],
            'pollToken' => ['required', 'string'],
        ]));
    }
}

File: examples/spring-boot/src/main/java/com/example/koma/web/KomaController.java
package com.example.koma.web;

import com.example.koma.service.KomaService;
import java.util.Map;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class KomaController {
  private final KomaService komaService;

  public KomaController(KomaService komaService) {
    this.komaService = komaService;
  }

  @PostMapping("/api/koma-qr")
  public Map<String, Object> qr(@RequestBody Map<String, String> body) {
    return komaService.createQrSession(body.get("amount"), body.get("currency"), body.get("productId"));
  }
}
